quickstart/101-app-service-backup/main.tf

```diff
@@ -0,0 +1,119 @@
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  location = var.resource_group_location
+  name     = random_pet.rg_name.id
+}
+
+resource "random_string" "storage_account_name" {
+  length  = 8
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "azurerm_storage_account" "example" {
+  name                     = random_string.storage_account_name.result
+  resource_group_name      = azurerm_resource_group.rg.name
+  location                 = azurerm_resource_group.rg.location
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "random_string" "storage_container_name" {
+  length  = 8
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "azurerm_storage_container" "example" {
+  name                  = random_string.storage_container_name.result
+  storage_account_name  = azurerm_storage_account.example.name
+  container_access_type = "private"
+}
+
+resource "random_string" "service_plan_name" {
+  length  = 8
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "azurerm_service_plan" "example" {
+  name                = random_string.service_plan_name.result
+  location            = azurerm_resource_group.rg.location
+  resource_group_name = azurerm_resource_group.rg.name
+  os_type             = "Windows"
+  sku_name            = "S1"
+}
+
+data "azurerm_storage_account_sas" "example" {
+  connection_string = azurerm_storage_account.example.primary_connection_string
+  https_only        = true
+
+  resource_types {
+    service   = false
+    container = false
+    object    = true
+  }
+
+  services {
+    blob  = true
+    queue = false
+    table = false
+    file  = false
+  }
+
+  start  = "2024-01-01"
```

Comment:
Would this hard code date be invalid from next year? Could we try `formatdate("YYYY-MM-DD", timestamp())` for `start`, and `formatdate("YYYY-MM-DD", timeadd(timestamp(), "8765h"))` for `expire`, and add a comment to tell the reader, this dynamic date is only for testing purpose, please do not follow in the production environment?

Reply:
Done